torch/fx/node.py

```diff
@@ -650,7 +650,11 @@ def is_impure(self):
 
         # Check if an impure function.
         if self.op == "call_function":
-            return self.target in _side_effectful_functions
+            if self.target in _side_effectful_functions:
+                return True
+            if isinstance(self.target, torch._ops.OpOverload) and self.target._schema.is_mutable:
```

Comment:
hmm this feels right, but... i wonder why we haven't just always set up FX's impure check to check schema. Maybe @angelayi or @zou3519 if you happen to know?

Reply:
I'll take a look! I think we should be able to add the check in #130552 (it check the schema) to the default DCE pass. Previously some tests were failing, but it should have been fixed now by #130598.

Reply:
sounds great! I will revert back this change in my PR and just do a point fix for the op we care about here. Would be great to check the schema in general :)